src/binary/ibc-setup/commands/connections.ts

```diff
@@ -0,0 +1,111 @@
+import { loadAndValidateApp } from '../../utils/load-and-validate-app';
+import { loadAndValidateRegistry } from '../../utils/load-and-validate-registry';
+import { resolveOption } from '../../utils/options/resolve-option';
+import { resolveHomeOption } from '../../utils/options/shared/resolve-home-option';
+import { resolveKeyFileOption } from '../../utils/options/shared/resolve-key-file-option';
+import { resolveMnemonicOption } from '../../utils/options/shared/resolve-mnemonic-option';
+import { signingClient } from '../../utils/signing-client';
+
+export type Flags = {
+  readonly home?: string;
+  readonly chain?: string;
+  readonly mnemonic?: string;
+  readonly keyFile?: string;
+  readonly interactive: boolean;
+};
+
+export type Options = {
+  readonly home: string;
+  readonly chain: string;
+  readonly mnemonic: string | null;
+};
+
+export async function connections(flags: Flags, logger: Logger) {
+  const home = resolveHomeOption({ homeFlag: flags.home });
+  const app = loadAndValidateApp(home);
+  const keyFile = resolveKeyFileOption({ keyFileFlag: flags.keyFile, app });
+  const chain = resolveOption('chain', { required: true })(
+    flags.chain,
+    process.env.RELAYER_CHAIN
+  );
+
+  const mnemonic = await resolveMnemonicOption(
+    {
+      interactiveFlag: flags.interactive,
+      mnemonicFlag: flags.mnemonic,
+      keyFile,
+      app,
+    },
+    true // mnemonic is optional
+  );
+
+  const options: Options = {
+    home,
+    chain,
+    mnemonic,
+  };
+
+  await run(options, logger);
+}
+
+function connectionStateAsText(state: ConnectionState) {
+  switch (state) {
+    case ConnectionState.STATE_INIT:
+      return 'Init';
+
+    case ConnectionState.STATE_OPEN:
+      return 'Open';
+
+    case ConnectionState.STATE_TRYOPEN:
+      return 'Tryopen';
+
+    case ConnectionState.STATE_UNINITIALIZED_UNSPECIFIED:
+      return 'UninitializedUnspecified';
+
+    case ConnectionState.UNRECOGNIZED:
+    default:
+      return 'Unrecognized';
+  }
+}
+
+export async function run(options: Options, logger: Logger) {
+  const registryFilePath = path.join(options.home, registryFile);
+  const registry = loadAndValidateRegistry(registryFilePath);
+
+  const chain = registry.chains[options.chain];
+  if (!chain) {
+    throw new Error(`Chain ${options.chain} not found in ${registryFilePath}.`);
+  }
+
+  const mnemonic = options.mnemonic ?? generateMnemonic();
+
+  const client = await signingClient(chain, mnemonic, logger);
+
+  const {
+    connections: allConnections,
+  } = await client.query.ibc.connection.allConnections();
+
+  const connections = allConnections.map((connection: IdentifiedConnection) => [
+    connection.id,
+    connectionStateAsText(connection.state),
+  ]);
+
+  if (!connections.length) {
+    logger.info(`No connections found for chain "${options.chain}".`);
+
+    return;
+  }
+
+  const output = borderLessTable([['CONNECTION_ID', 'STATE'], ...connections]);
```

Comment:
Do we want to expose any other information? Currently, it looks like this:
<img width="184" alt="image" src="https://user-images.githubusercontent.com/79154001/111454031-7f505980-8714-11eb-8e7f-965612ea5852.png">

Available data:
<img width="120" alt="image" src="https://user-images.githubusercontent.com/79154001/111454087-8d05df00-8714-11eb-8258-5e315fe57a7e.png">

Reply:
Those are useful infos.

It would be nice to show the clientId... or actually info on the client (dig down to the chainId given that clientID). But I would add that one later as another issue (TODO: write issue).

delayPeriod should be 0 for all the ones we create (hardcoded somewhere). It is very important to show that if non-zero, but we don't support it yet https://github.com/confio/ts-relayer/issues/48 A follow up issue to allow us to configure the delay on creation (but only use 0 in all CLI) and then showing that value if present would be useful.

Reply:
For now, just show Delay and ClientID in new columns and we can improve that later. (Delay should always be 0 here, add a test when we can set it).

Reply:
682cc17